src/java/org/apache/cassandra/db/memtable/Memtable.java

```diff
@@ -0,0 +1,389 @@
+         * instead by returning true here.
+         */
+        default boolean streamToMemtable()
+        {
+            return false;
+        }
+
+        /**
+         * When we need to stream data, we usually flush and stream the resulting sstables. If the sstable does not
+         * want to flush, it should return true here and false on shouldSwitch(STREAMING/REPAIR).
+         * In that case, temporary sstables will be created, streamed and then deleted.
+         */
+        default boolean streamFromMemtable()
+        {
+            return false;
+        }
+    }
+
+    /**
+     * Interface for providing signals back to the owner.
+     */
+    interface Owner
+    {
+        void signalFlushRequired(Memtable memtable, ColumnFamilyStore.FlushReason reason);
+    }
+
+
+    // Main write and read operations
+
+    /**
+     * Put new data in the memtable. This operation may block until enough memory is available in the memory pool.
+     *
+     * @param update the partition update, may be a new partition or an update to an existing one
+     * @param indexer receives information about the update's effect
+     * @param opGroup write operation group, used to permit the operation to complete if it is needed to complete a
+     *                flush to free space.
+     *
+     * @return the smallest timestamp delta between corresponding rows from existing and update. A
+     * timestamp delta being computed as the difference between the cells and DeletionTimes from any existing partition
+     * and those in {@code update}. See CASSANDRA-7979.
+     */
+    long put(PartitionUpdate update, UpdateTransaction indexer, OpOrder.Group opGroup);
+
+    /**
+     * Get the partition for the specified key. Returns null if no such partition is present.
+     */
+    Partition getPartition(DecoratedKey key);
+
+    /**
+     * Returns a partition iterator for the given data range.
+     *
+     * @param columnFilter filter to apply to all returned partitions
+     * @param dataRange the partition and clustering range queried
+     */
+    MemtableUnfilteredPartitionIterator makePartitionIterator(ColumnFilter columnFilter,
+                                                              DataRange dataRange);
+
+    interface MemtableUnfilteredPartitionIterator extends UnfilteredPartitionIterator
+    {
+        /**
+         * Returns the minimum local deletion time for all partitions in the range.
+         * Required for the efficiency of partition range read commands.
+         */
+        int getMinLocalDeletionTime();
+    }
+
+
+    // Statistics
+
+    /** Number of partitions stored in the memtable */
+    int partitionCount();
+
+    /** Size of the data not accounting for any metadata / mapping overheads */
+    long getLiveDataSize();
+
+    /** Number of operations (i.e. {@link #put} calls) the memtable has executed */
+    long getOperations();
```

Comment:
can it be `getNumPuts` or `getNumWrites` or sth along these lines? `Operations` is very generic and thus it requries to read the comment or implementation to know what is the semantics

Reply:
I prefer to leave this murky, to be honest, to ensure no one starts trusting a clearer name. It doesn't tend to be what one expects (e.g. a write with a static column is counted as two operations). It's also present in this form user-visible messages.

Let me know if you insist on changing it.

Reply:
OK, I see that operation is defined in `PartitionUpdate::operationCount()`
might be worth mentioning (or not - up to you)